packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -60,15 +65,26 @@ class ConfigChainBuilder {
     while (loc !== (loc = path.dirname(loc))) {
       if (!foundConfig) {
         const configLoc = path.join(loc, BABELRC_FILENAME);
-        if (exists(configLoc)) {
-          this.addConfig(configLoc);
-          foundConfig = true;
-        }
-
+        const configJSLoc = path.join(loc, BABELRC_JS_FILENAME);
         const pkgLoc = path.join(loc, PACKAGE_FILENAME);
-        if (!foundConfig && exists(pkgLoc)) {
-          foundConfig = this.addConfig(pkgLoc, "babel", JSON);
-        }
+        const configLocs = [configLoc, configJSLoc, pkgLoc];
+        const foundConfigs = configLocs.reduce((arr, config) => {
+          if (exists(config)) {
+            const configAdded = config === pkgLoc
+              ? this.addConfig(config, "babel", JSON)
+              : this.addConfig(config);
+
+            if (configAdded && arr.length) {
```

Comment:
Perfect, we should rebase this PR onto the 7.0 branch then, it's currently on `master` which threw me off.